usr/sbin/rear

```diff
@@ -50,23 +68,34 @@ function apply_bash_flags_and_options_commands () {
 #   apply_bash_flags_and_options_commands "$INITIAL_BASH_FLAGS_AND_OPTIONS_COMMANDS"
 readonly INITIAL_BASH_FLAGS_AND_OPTIONS_COMMANDS="$( get_bash_flags_and_options_commands )"
 
-# Enable as needed for development of fail-safe code:
-# set -ue -o pipefail
-# set -xue -o pipefail
-# set -xvue -o pipefail
-# Cf. the Relax-and-Recover Coding Style
-# https://github.com/rear/rear/wiki/Coding-Style
-# that reads (excerpt - dated 18. Nov. 2015):
-# "TODO Use set -eu to die from unset variables and unhandled errors"
-
-# Versioning
-readonly PRODUCT="Relax-and-Recover"
-readonly PROGRAM=${0##*/}
-readonly VERSION=2.9
-readonly RELEASE_DATE="2025-01-31"
-
-# Where users should report bugs:
-readonly BUG_REPORT_SITE="https://github.com/rear/rear/issues"
+# To be on the safe side forbid careless/insecure sourcing
+# via plain 'source' and '.' from the very beginning
+# see in general https://relax-and-recover.org/documentation/security-architecture
+# and see in particular https://github.com/rear/rear/issues/3259
+# and https://github.com/rear/rear/issues/3259#issuecomment-2385745545
+# and https://github.com/rear/rear/issues/3319#issuecomment-2360546784
+# and https://github.com/rear/rear/issues/3319#issuecomment-2363556217
+function . () {
+    echo -e "ERROR: BUG in $PRODUCT\nForbidden usage of '.' in\n  . $*\nPlease report it at $BUG_REPORT_SITE" >&2
+    exit 1
+}
+# To be on the safe side sourcing via '.' is kept always forbidden in ReaR because
+# it is basically impossible to reliably find all code places where '.' is used for sourcing
+# see https://github.com/rear/rear/issues/3260#issuecomment-2248252572
+# In contrast to sourcing via '.' which is kept forbidden
+# for sourcing via 'source' later (see below) trusted sourcing gets set up
+# by replacing this 'source' forbid function by a 'source' wrapper function
+# in lib/_framework-setup-and-functions.sh which implements trusted sourcing
+# which is then the only allowed way to normally source a file in ReaR.
+# For exceptional cases 'builtin source' can still be used to source files
+# that are known to be trusted (e.g. with hardcoded trusted path).
+# The point is that we can reliably find all code places where 'builtin' is used
+# so we can reliably check such exceptional cases at any time to keep ReaR secure
+# (in particular we can also find where the undesirable form 'builtin .' is used):
+function source () {
+    echo -e "ERROR: BUG in $PRODUCT\nForbidden usage of 'source' in\n  source $*\nPlease report it at $BUG_REPORT_SITE" >&2
```

Comment:
```suggestion
    echo -e "ERROR: BUG in $PRODUCT\nForbidden usage of Bash 'source' instead of ReaR 'Source' in\n  source $*\nPlease report it at $BUG_REPORT_SITE" >&2
```

Reply:
It is not about using ReaR's 'Source' function
because this function is only to source ReaR files
(ReaR scripts or ReaR config files).

It is about not using 'source'
until the source wrapper function in
_framework-setup-and-functions.sh is defined,
see my above
https://github.com/rear/rear/pull/3424/files#r1992949757